### PR TITLE
CDAP-1456 CLI should have the ability to create Dataset instance with Properties

### PR DESCRIPTION
**Pull request by jwang47**

### CODE REVIEW

**Comment by jwang47** [comment]
Build running http://builds.cask.co/browse/CDAP-DUT810-1

**Comment by jwang47** [comment]
Build running http://builds.cask.co/browse/CDAP-DUT810-2

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/CreateDatasetInstanceCommand.java [hunk 1]
```diff
@@ -43,14 +49,20 @@ public CreateDatasetInstanceCommand(DatasetClient datasetClient, CLIConfig cliCo
...
+    String datasetPropertiesString = arguments.get(ArgumentName.DATASET_PROPERTIES.toString(), "");
+    Map<String, String> datasetProperties = ArgumentParser.parseMap(datasetPropertiesString);
+    DatasetInstanceConfiguration datasetConfig = new DatasetInstanceConfiguration(datasetType, datasetProperties);
 
-    datasetClient.create(datasetName, datasetType);
-    output.printf("Successfully created dataset named '%s' with type '%s'\n", datasetName, datasetType);
+    datasetClient.create(datasetName, datasetConfig);
+    output.printf("Successfully created dataset named '%s' with type '%s' and properties '%s'",
+                  datasetName, datasetType, GSON.toJson(datasetProperties));
+    output.println();
```
Comment (ParthGandhi) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/CreateDatasetInstanceCommand.java]: Why new line?

**Comment by ParthGandhi** [comment]
LGTM

**Comment by jwang47** [comment]
Build passed, merging http://builds.cask.co/browse/CDAP-DUT810-3

**Pull request merged**